sale_fixed_discount/views/sale_portal_templates.xml

```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template
+        id="sale_order_portal_content"
+        inherit_id="sale.sale_order_portal_content"
+        priority="100"
+    >
+        <xpath
+            expr="//section[@id='details']//t[@t-set='display_discount']"
+            position="after"
+        >
+            <t
+                t-set="display_discount_fixed"
+                t-value="True in [line.discount_fixed > 0 for line in sale_order.order_line]"
+            />
+        </xpath>
+        <xpath
+            expr="//section[@id='details']//table/thead//th[@t-if='display_discount']"
+            position="before"
+        >
+            <th
+                t-if="display_discount_fixed"
+                t-attf-class="text-end {{ 'd-none d-sm-table-cell' if report_type == 'html' else '' }}"
+            >
+                <span>Disc. Fixed Amount</span>
+            </th>
+        </xpath>
+        <xpath
+            expr="//section[@id='details']//tbody[hasclass('sale_tbody')]//t[@t-foreach='lines_to_report']/tr//t[@t-if='not line.display_type']//td[3]/div"
```

Comment:
I get the following error when I install the module from your branch.

```
Traceback (most recent call last):
  File "/src/odoo/odoo/http.py", line 1584, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/src/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/src/odoo/odoo/http.py", line 1611, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/src/odoo/odoo/http.py", line 1815, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/src/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/src/odoo/odoo/http.py", line 697, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/dataset.py", line 46, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/src/odoo/odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/src/odoo/odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-74>", line 2, in button_immediate_install
  File "/src/odoo/odoo/addons/base/models/ir_module.py", line 74, in check_and_log
    return method(self, *args, **kwargs)
  File "/src/odoo/odoo/addons/base/models/ir_module.py", line 456, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/src/odoo/odoo/addons/base/models/ir_module.py", line 580, in _button_immediate_function
    registry = modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "<decorator-gen-14>", line 2, in new
  File "/src/odoo/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/src/odoo/odoo/modules/registry.py", line 90, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/src/odoo/odoo/modules/loading.py", line 488, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/src/odoo/odoo/modules/loading.py", line 372, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/src/odoo/odoo/modules/loading.py", line 231, in load_module_graph
    load_data(cr, idref, mode, kind='data', package=package)
  File "/src/odoo/odoo/modules/loading.py", line 71, in load_data
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind)
  File "/src/odoo/odoo/tools/convert.py", line 763, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate)
  File "/src/odoo/odoo/tools/convert.py", line 829, in convert_xml_import
    obj.parse(doc.getroot())
  File "/src/odoo/odoo/tools/convert.py", line 749, in parse
    self._tag_root(de)
  File "/src/odoo/odoo/tools/convert.py", line 709, in _tag_root
    raise ParseError(msg) from None  # Restart with "--log-handler odoo.tools.convert:DEBUG" for complete traceback
odoo.tools.convert.ParseError: while parsing None:7
Error while validating view:

Element '<xpath expr="//section[@id=&#39;details&#39;]//table/tbody//t[@t-foreach=&#39;sale_order.order_line&#39;]/tr//t[@t-if=&#39;not line.display_type&#39;]//td[3]/div">' cannot be located in parent view

View error context:
{'file': '/src/user/modules/oca/sale-workflow/sale_fixed_discount/views/sale_portal_templates.xml',
 'line': 2,
 'name': 'sale_order_portal_content_fixed_discount',
 'view': ir.ui.view(870,),
 'view.model': False,
 'view.parent': ir.ui.view(861,),
 'xmlid': 'sale_order_portal_content_fixed_discount'}
```

can't we change it to:

```
        <xpath
            expr="//div[@t-field='line.price_unit']"
            position="replace"
        >
            <div
                t-if="line.discount &gt;= 0 or line.fixed_discount &gt;= 0"
                t-field="line.price_unit"
                t-att-style="(line.discount or line.discount_fixed) and 'text-decoration: line-through' or None"
                t-att-class="((line.discount or line.discount_fixed) and 'text-danger' or '') + ' text-right'"
            />
            <div t-if="line.discount_fixed">
                <t
                    t-esc="line.price_unit - line.discount_fixed"
                    t-options='{"widget": "float", "decimal_precision": "Product Price"}'
                />
            </div>
        </xpath>
```

Reply:
There's a warning from pylint-odoo when you use `replace`. I specifically had to refactor this in order to not get the warning.

Reply:
I wasn't happy using replace either. Something which should be avoided so modules do not overwrite each other. But I did not have a good idea in this case. 

To bypass this warning you can set a priority="100" in the template node.

I'll have a look on your solution.

Reply:
Please disregard my previous comment. My assumption was wrong. I did however test the module again and did not come across your installation errors. 
I ran it on a new demo database and only installed `sale_fixed_discount` and manually added `sale_management`. Nothing more. 
Are you experiencing a side effect from another module in your installation?